### PR TITLE
Added the init file to start/reload/stop Galaxy with uWSGI and job ha…

### PR DESCRIPTION
**Pull request by remimarenco**
…ndlers

Hi!

As we are using uWSGI and Job Handlers on our Galaxy instance, we needed to have a script to start/reload/stop.
Here is a first version we wanted to share with you :).

This script is using the following cmds:
- `PYTHONPATH=eggs/PasteDeploy-1.5.0-py2.7.egg uwsgi --ini-paste config/galaxy.ini --master-fifo masterFifo &` as stated [here](https://wiki.galaxyproject.org/Admin/Config/Performance/Scaling#line-92)
- `GALAXY_RUN_ALL=1 sh run.sh --daemon`, `GALAXY_RUN_ALL=1 sh run.sh --stop-daemon` and `GALAXY_RUN_ALL=1 ./run.sh --wait restart` from the [Galaxy documentation](https://wiki.galaxyproject.org/Admin/Config/Performance/Scaling#Starting_and_Stopping)
- `echo r > masterFifo` and `echo q > masterFifo` from the [uWSGI documentation](http://uwsgi-docs-additions.readthedocs.org/en/latest/MasterFIFO.html) with MasterFIFO file

Configure the `RUN_AS="galaxy"` and the `RUN_IN="/path/to/galaxy-dist"` to match your installation, and put this file to the location you are managing your apps (usually /etc/init.d).

### CODE REVIEW

**Comment by nsoranzo** [comment]
It seems the file is mixing tabs and spaces for indentation, can you convert all tabs to spaces?

**Comment by hexylena** [comment]
@remimarenco I feel like most people using uWSGI for Galaxy are also using Supervisord to manage it. Supervisord is far, far superior to the old way of having shell scripts to manage as it can do things like automatically restarting the processes in case of failures.

File: contrib/galaxy.fedora.uwsgi.handlers-init [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+start() {
+	echo "Starting $SERVICE_NAME... "
+	cmd_uwsgi="cd $RUN_IN && PYTHONPATH=eggs/PasteDeploy-1.5.0-py2.7.egg uwsgi --ini-paste config/galaxy.ini --master-fifo masterFifo &"
+        cmd_galaxy_daemons="cd $RUN_IN && GALAXY_RUN_ALL=1 sh run.sh --daemon"
+	case "$(id -un)" in
+		$RUN_AS)
+			eval "$cmd_uwsgi"
+			eval "$cmd_galaxy_daemons"
+			;;
+		root)
```
Comment (nsoranzo) [inline contrib/galaxy.fedora.uwsgi.handlers-init]: You don't probably need a special case for `$RUN_AS`, you can use the same commands below.

File: contrib/galaxy.fedora.uwsgi.handlers-init [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+			echo "*** ERROR *** must be $RUN_AS or root in order to control this service" >&2
+			exit 1
+	esac
+	
+	echo "done."
+}
+
+reload() {
+	echo -n "Reloading $SERVICE_NAME... "
+	cmd_uwsgi="cd $RUN_IN && (echo r > masterFifo)"
```
Comment (nsoranzo) [inline contrib/galaxy.fedora.uwsgi.handlers-init]: You are not restarting the handlers here. You can use:

```
GALAXY_RUN_ALL=1 ./run.sh --wait restart
```

**Comment by remimarenco** [comment]
Thanks for all your feedbacks, I will modify this as soon as possible :D

File: contrib/galaxy.fedora.uwsgi.handlers-init [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+SERVICE_NAME="galaxy"
+RUN_AS="galaxy"
+RUN_IN="/path/to/galaxy-dist"
+
+#--- main actions
+
+start() {
+    echo "Starting $SERVICE_NAME... "
+    cmd_uwsgi="cd $RUN_IN && PYTHONPATH=eggs/PasteDeploy-1.5.0-py2.7.egg uwsgi --ini-paste config/galaxy.ini --master-fifo masterFifo &"
```
Comment (jmchilton) [inline contrib/galaxy.fedora.uwsgi.handlers-init:25]: This is probably no longer correct since the eggs stuff has disappeared.

**Comment by jmchilton** [comment]
Marking this a WIP until the egg issue and other issues are addressed.

**Comment by jmchilton** [comment]
Got @remimarenco's permission to close this for now.

**Pull request closed without merging**